Unit tests failing as `setup-envtest.sh` from kubernetes-sigs/controller-runtime now returns a 404

In the GH action run of our unit tests there was a sudden [failure](https://github.com/k8ssandra/k8ssandra/runs/2480849950).

```console
o fmt ./pkg/...
pkg/cleaner/cleaner.go
pkg/cleaner/cleaner_test.go
go fmt ./tests/...
go vet ./pkg/...
go vet ./tests/...
go test  -test.timeout=3m ./tests/unit/... -coverprofile cover.out
ok  	github.com/k8ssandra/k8ssandra/tests/unit	76.584s	coverage: [no statements]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/cassdc	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/helm	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/kubeapi	[no test files]
?   	github.com/k8ssandra/k8ssandra/tests/unit/utils/traefik	[no test files]
mkdir -p /home/runner/work/k8ssandra/k8ssandra/testbin
test -f /home/runner/work/k8ssandra/k8ssandra/testbin/setup-envtest.sh || curl -sSLo /home/runner/work/k8ssandra/k8ssandra/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/master/hack/setup-envtest.sh
source /home/runner/work/k8ssandra/k8ssandra/testbin/setup-envtest.sh && fetch_envtest_tools /home/runner/work/k8ssandra/k8ssandra/testbin && setup_envtest_env /home/runner/work/k8ssandra/k8ssandra/testbin && go test  -test.timeout=3m ./pkg/... -coverprofile cover.out
/home/runner/work/k8ssandra/k8ssandra/testbin/setup-envtest.sh: line 1: 404:: command not found
make: *** [Makefile:35: pkg-test] Error 127
Error: Process completed with exit code 2.
```

Looking at the logic our `curl` of `https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/master/hack/setup-envtest.sh` returns

```console
404: Not Found
```

--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ endif
 
 pkg-test:
 	mkdir -p ${ENVTEST_ASSETS_DIR}
-	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/master/hack/setup-envtest.sh
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
 	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh && fetch_envtest_tools $(ENVTEST_ASSETS_DIR) && setup_envtest_env $(ENVTEST_ASSETS_DIR) && go test $(GO_FLAGS) -test.timeout=3m ./pkg/... -coverprofile cover.out
 
 integ-test:

